app/js/registration_controller.js

```diff
@@ -16,13 +44,24 @@
     Utils.toggle_spinner();
 
     promise.then(function() {
-      var success = document.getElementById('success_message');
       var success_email = document.getElementById('success_email');
       success_email.appendChild(document.createTextNode(email));
       form.classList.add('hide');
-      success.classList.remove('hide');
+      success_msg.classList.remove('hide');
       Utils.toggle_spinner();
-    }).catch(function(response) {
+
+      poll_for_email_confirmation();
+
+      document.addEventListener('visibilitychange', function() {
```

Comment:
We should only check for email confirmation when `!document.hidden`. No need to fire an HTTP request when we go off screen.